ecommerce/views.py

```diff
@@ -40,3 +39,110 @@ def update_profile(request):
         profile_form = ProfileForm(instance=request.user.profile)
     return render(request, 'ecommerce/profile.html', {'user_form': user_form,'profile_form': profile_form})
+
+@login_required
+@transaction.atomic
+def cart_add(request, pk):
+    if request.method == 'POST':
+        quantity = request.POST['quantity']
+    else:
+        quantity = 0
+
+    product_get = Product.objects.filter(pk=pk).first()
+    if product_get:
+        product_stock = product_get.quantity
+    else:
+        product_stock = 0
+
+    in_cart = Cart.objects.filter(user=request.user, product__pk=pk).first()
+    if in_cart:
+        cart_quantity = in_cart.quantity
+    else:
+        cart_quantity = 0
+    
+    if int(quantity) > 0 :
+        if int(quantity) + cart_quantity > product_stock:
+            messages.add_message(request, messages.INFO, _("Out of stock limit"))
+        else:
+            product_temp = Cart.objects.filter(user=request.user, product__pk=pk).first()
+            if product_temp:
+                pd_temp = product_temp.get(product__pk=pk)
+                pd_temp.quantity = int(quantity) + cart_quantity
+                pd_temp.save()
+            else:
+                product_new = Cart(user=request.user, product=Product.objects.get(pk=pk), quantity=quantity)
+                product_new.save()
+            messages.add_message(request, messages.INFO, _('You added a product to your cart'))
+    else:
+        messages.add_message(request, messages.INFO, _('Please add product quantity more than 0'))
+    
+    return redirect(f'/ecommerce/product/{pk}')
+
+@login_required
+def cart_get(request):
+    cart = Cart.objects.filter(user=request.user).select_related('product')
+    stock_changed = False
+    total_price = 0
+    for item in cart:
+        in_stock = item.product.quantity
+        if in_stock == 0:
+            Cart.objects.filter(user=request.user, product=item.product).delete()
+            stock_changed = True
+        elif item.quantity > in_stock:
+            messages.add_message(request, messages.INFO, _(f"{item.product.product_name} is out of stock limit, please decrease your product items"))
+            stock_changed = True
+        total_price += item.product.price * item.quantity
+
+    return render(request, 'ecommerce/cart.html', {'cart': cart, 'changed':stock_changed, 'total':total_price})
+
+@login_required
+def increase_product_in_cart(request, pk):
+    product_get = Product.objects.filter(pk=pk).first()
+    if product_get:
+        product_stock = product_get.quantity
+    else:
+        product_stock = 0
+
+    item = Cart.objects.filter(user=request.user, product__pk=pk).first()
+    if item:
+        cart_quantity = item.quantity
+    else:
+        cart_quantity = 0
+    
+    if cart_quantity + 1 > product_stock:
+        messages.add_message(request, messages.INFO, _("Out of stock limit"))
+    else:
+        pd_temp = item
+        pd_temp.quantity = cart_quantity + 1
+        pd_temp.save()
```

Comment:
dùng trực tiếp item đã có rồi đi em
```suggestion
        item.quantity = cart_quantity + 1
        item.save()
```